chore: add currency information and local names to country list

abdullahceylan opened the pull request:
This PR enhances the country list by adding comprehensive currency information and local names for all countries. The additions include:

- Local names in native languages/scripts (solves #12)
- Currency names
- Currency codes (ISO 4217)
- Currency symbols

## Changes Made
- Updated `CountryInterface` type to include new optional fields:
  - `currency?: string` - Full currency name
  - `currency_code?: string` - ISO 4217 currency code
  - `currency_symbol?: string` - Currency symbol

- Added currency information for all countries
- Fixed corrupted flag emojis for several countries
- Updated local names to be more accurate and include native scripts
- Standardized currency symbol formats

### Sources
- **Currency information:** https://www.xe.com/en-gb/symbols 
- **Local names:** https://worldpopulationreview.com/country-rankings/names-of-countries-in-their-own-language

Skarvion commented:
I'm not the maintainer, but looking at this, there are few feedbacks:

- It will be great if the data can work with https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat, but we'll need the locale information which we don't have
- Probably worth adding a test case that ensure the currency format in the `data.ts` is the same with the `Intl.NumberFormat` content

Cheers for the PR!

necessarylion commented:
Thanks a bunch for the PR! It looks great to me.

By the way, could you please add the newly added keys function to the `src/country.ts` file?

Also, make sure you have a cover test case for it. Thanks again!

dannya commented:
Thanks for this work, I'm very interested in getting this PR merged!

In the interests of saving data, since the `localName` accessor already returns `local_name` if set, then falls back to `name` if it is not set:
![image](https://github.com/user-attachments/assets/8de9c891-e6c7-4caa-ab5a-a70041a7c194)

...can `local_name` be omitted in data.ts unless it is actually different than the `name`?

Happy to do this work myself if it helps get this merged!

abdullahceylan commented:
> I'm not the maintainer, but looking at this, there are few feedbacks:
> 
> * It will be great if the data can work with https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat, but we'll need the locale information which we don't have
> * Probably worth adding a test case that ensure the currency format in the `data.ts` is the same with the `Intl.NumberFormat` content
> 
> Cheers for the PR!

Thanks for the feedback @Skarvion. I added the test cases.


> Thanks a bunch for the PR! It looks great to me.
> 
> By the way, could you please add the newly added keys function to the `src/country.ts` file?
> 
> Also, make sure you have a cover test case for it. Thanks again!

Thanks @necessarylion. I added the helper functions.


> Thanks for this work, I'm very interested in getting this PR merged!
> 
> In the interests of saving data, since the `localName` accessor already returns `local_name` if set, then falls back to `name` if it is not set: ![image](https://private-user-images.githubusercontent.com/394687/407847886-8de9c891-e6c7-4caa-ab5a-a70041a7c194.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgyNDg4NzMsIm5iZiI6MTczODI0ODU3MywicGF0aCI6Ii8zOTQ2ODcvNDA3ODQ3ODg2LThkZTljODkxLWU2YzctNGNhYS1hYjVhLWE3MDA0MWE3YzE5NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEzMFQxNDQ5MzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYzE4N2I5ZmUyZmU0ZjVkMzA2OGJmMjk0OTQ5OTI3YTYwYTE5NWQ2ZjYyNmEyN2M5ZTdlNTA4YmMxNjk2NTljJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.I4kb1Ic-fjATkA-9I3sqChSwUyIsH2DUwZkG3qt-4eM)
> 
> ...can `local_name` be omitted in data.ts unless it is actually different than the `name`?
> 
> Happy to do this work myself if it helps get this merged!

Thanks for your feedback @dannya. I mentioned the reference links in the PR description but additionally, you can check https://www.worldatlas.com/geography/names-of-countries-in-their-own-languages.html for the differences.

dannya commented:
@abdullahceylan thanks - I'm referring to the items in `data.ts` where the `name` is the same as `local_name`, like:

![image](https://github.com/user-attachments/assets/82b96dd4-1e90-4ac2-b9b9-4ad17141656f)

In these cases, there is no need to define a `local_name`.

necessarylion commented on src/data.ts:
```diff
@@ -1,344 +1,2603 @@
+    local_name: 'Australia',
+    currency: 'Australian Dollar',
+    currency_code: 'AUD',
+    currency_symbol: '$',
```
May be better to use like `A$` for all the `XXX Dollar`?
Otherwise it will be same as USD symbol.

abdullahceylan replied:
I believe this is the most common one. `Intl.NumberFormat` also returns the $ symbol.

necessarylion commented on __tests__/currency.test.ts:
```diff
@@ -0,0 +1,67 @@
+
+      // Verify that the currency code matches what Intl.NumberFormat expects
+      expect(() => formatter.format(testAmount)).not.toThrow()
+      expect(country?.currencyCode).toBe(currency)
```
Can we also verify that the symbol match with Intl.NumberFormat please? I am interested to see if currency like `AUD` match with `A$` or just `$`.

abdullahceylan commented:
> @abdullahceylan thanks - I'm referring to the items in `data.ts` where the `name` is the same as `local_name`, like:
> 
> ![image](https://private-user-images.githubusercontent.com/394687/408179495-82b96dd4-1e90-4ac2-b9b9-4ad17141656f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzgyNTQ5NDEsIm5iZiI6MTczODI1NDY0MSwicGF0aCI6Ii8zOTQ2ODcvNDA4MTc5NDk1LTgyYjk2ZGQ0LTFlOTAtNGFjMi1iOWI5LTRhZDE3MTQxNjU2Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEzMFQxNjMwNDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZjJlN2ZjOGFiZWJmYzNkODcxMmRiMDZmNTY3OTQ2Njc4MmY0YjQwYzhmYWQ1Mjc2MzRmOGEwOGJiMmU5OWFhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._3_7I9vJfRQeblKsYYTiABrhys0JI4fy-0iZnchvl64)
> 
> In these cases, there is no need to define a `local_name`.

I think it can be done in a separate PR if needed as don't think there are many cases like that.
